tonic/src/transport/server/conn.rs

```diff
@@ -98,6 +98,34 @@ impl Connected for TcpStream {
     }
 }
 
+/// Connection info for Unix domain socket streams.
+///
+/// This type will be accessible through [request extensions][ext] if you're using
+/// a unix stream.
+///
+/// See [`Connected`] for more details.
+///
+/// [ext]: crate::Request::extensions
+#[cfg(unix)]
+#[cfg_attr(docsrs, doc(cfg(unix)))]
+#[derive(Clone, Debug)]
+pub struct UdsConnectInfo {
+    peer_addr: Option<Arc<tokio::net::unix::SocketAddr>>,
+    peer_cred: Option<tokio::net::unix::UCred>,
```

Comment:
Uhhh, this made me realize I'm being a bonehead here 🤦🏽  Stay tuned

Reply:
@Spferical Well, this made me dig in pretty deep here and I realized that the case for unix sockets is a bit weird. It doesn't seem to matter whether or not the `tls` feature is disabled or not - the `examples` use the `tls` feature while the integration tests do not - but in both cases the examples/tests work as expected.

I'm not sure if the tonic folks would say:
1) 🤷🏽 "Yeah, messy, whatever, it works"
2) 👎🏽 MESSY, let's create some new methods `serve_with_incoming_unix`/`serve_with_incoming_shutdown_unix`/etc etc
3) something else?

Because of this, I'm going to close this PR and start a conversation with the tonic folks based off my findings for some advice on how to proceed/if they're willing to accept the patch.

Reply:
@Spferical Still not really sure on the above since there was no knee-jerk reaction from the [tonic maintainer who said they'd accept a PR for this](https://github.com/hyperium/tonic/issues/856#issuecomment-987425092).

I'm thinking I just fire off something like this to upstream tonic and see what they think. It's totally possible that this is fine as-is and we don't need to make a distinction between tcp and unix sockets at the top level. I think I'd rather leave as-is and see what they have to say. Thoughts?